common/lib/api-client/rest-client.js

```diff
@@ -0,0 +1,26 @@
+const axios = require('axios')
+
+const {
+  API: { BASE_URL },
+} = require('../../../config')
+
+const auth = require('./auth')()
+const getRequestHeaders = require('./request-headers')
+
+const restClient = async (url, params, options = {}) => {
```

Comment:
I wonder if there is a simpler way to achieve this for the CSV response. Could we not patch the [response middleware](https://github.com/twg/devour/blob/3ce573bb708d851069a3c6c9f6805fbad4adf3bb/src/middleware/json-api/res-deserialize.js#L12), in a similar way to how we currently [patch the post middleware](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/2721f4ecfb2590dd59f33a0a2299f2fc05039789/common/lib/api-client/middleware/post.js#L4) and just check the `Content-Type` header? If so, just return the data instead of deserializing.

Reply:
I tried - I really did. I don't see how you can skip the deserializing.

But I also think that it's cleaner to have a restful client for thing that the API exposes in a restful way.

Maybe the csv or data  should be a model type and we could return that as a property of the returned object for a json-api call